### ISSUE TITLE
Que

### ISSUE THREAD

**Issue by wry123456** [xiaopanliang/brushstroke-style-recognition#4]
Does this have a specific data set？

Is there a corresponding running result for the brushstroke features in the image?

**Comment by yahuiliang** [comment]
@wry123456 
Does this have a specific data set？
Yes, we downloaded the dataset called WikiArt. You can google it and download it. We did brushstroke extraction to images in this dataset.

Is there a corresponding running result for the brushstroke features in the image?
Yes, we have. This repo is incomplete. More information will be added to README in the future. The idea of the algorithm was originally from this [paper](http://infolab.stanford.edu/~wangz/project/imsearch/ART/PAMI11/li.pdf). Here is the example of the brushstroke extraction result:
![ori_img](https://user-images.githubusercontent.com/31673419/55350801-87108800-5482-11e9-9756-a111ad37fa9e.png)
![brushstroke](https://user-images.githubusercontent.com/31673419/55350825-92fc4a00-5482-11e9-9159-b4d1a81e6871.png)

**Comment by wry123456** [comment]
@yahuiliang 
Thank you for your reply.Yes,  I also got the corresponding results.
Can we segment the extracted brushstroke features by some methods in the library and use it better for feature learning?

**Comment by yahuiliang** [comment]
@wry123456 Thats a great question. However, we also feel confused about what can be extracted from brushstrokes. We had the hard time on understanding how the neural network learn remembering brushstrokes. If we can segment the brushstroke, we believe it can increase the accuracy of the neural network. The current neural network we are using is CNN. We feed the image data into the neural network directly, and we are not sure how CNN understands it.

**Comment by wry123456** [comment]
@yahuiliang Thank you for your reply, whether you can upload the cnn model and training files, or send them by email. w_ang_ruoyu@163.com

**Comment by wry123456** [comment]
@yahuiliang I have a serious problem when I use the cnn model to learn features. My model suddenly drops to 0 after iterating several times during training. I don’t know what caused it.
Model：(conv1): Sequential(
    (0): Conv2d(3, 20, kernel_size=(5, 5), stride=(1, 1))
    (1): BatchNorm2d(20, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): LeakyReLU(negative_slope=0.01)
  )
  (maxpool1): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  (conv2): Sequential(
    (0): Conv2d(20, 36, kernel_size=(5, 5), stride=(1, 1))
    (1): BatchNorm2d(36, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
    (2): LeakyReLU(negative_slope=0.01)
  )
  (maxpool2): MaxPool2d(kernel_size=2, stride=2, padding=0, dilation=1, ceil_mode=False)
  (fc): Linear(in_features=6084, out_features=300, bias=True)
)

Error：loss 1.9180773961124942e-05

loss 1.9311905816721264e-06

loss 2.3841858265427618e-08

loss 0.0

loss 0.0

loss 0.0
Please  give me some advice！

**Comment by yahuiliang** [comment]
@wry123456 Sorry for the late response. I just pushed our latest version CNN model the repo. It is under style_recognition.py file. However, our research is still in progress. The model it self is incomplete, but it is able to be trained.

**Comment by yahuiliang** [comment]
@wry123456 Based on what you show there, if the loss drops to zero, I think it means the model cannot be trained further. I recommend you to test the accuracy of the model to see how the result looks like. If the accuracy is low, like 50% or 60%, you should consider to modify the model to make it train further.
